poetry/solution_providers/__init__.py

```diff
@@ -0,0 +1,3 @@
+from .python_requirement_incompatibility_solution_provider import (
```

Comment:
These modules are a mouth full :) Would be great to put them into a structure, eg `poetry.solutions.providers.incompatibility.python_requreiment`. 

Would `solutions` be better placed in `poetry.mixology` consdiering we only really provide solutions for version solving issues or is the intent here to provide a generic mechanism?

Reply:
I wanted them to be as explicit as possible :-) But yeah I can see if we can reorganize a bit.

The idea is to extend the solutions beyond solver issues, for instance for configuration and validation errors.

Reply:
So, I reorganized a bit and now each module (in this case `mixology`) can provide its own solutions and providers.

This allows for a more flexible implementation.